.travis.yml

```diff
@@ -0,0 +1,16 @@
+sudo: false
+language: go
+go:
+- 1.7
+- 1.8
+- 1.9
+- tip
+install:
+- go get github.com/gogo/protobuf/...
+- go get github.com/golang/protobuf/...
+- go get github.com/pkg/errors/...
+- go get github.com/stretchr/testify/...
+- go get golang.org/x/net/...
+script:
+- go install ./...
+- go test -race ./...
```

Comment:
I'd like this to be `make test_all` so we run tests of client compatibility. But getting there would probably take a bunch more work - we'd need to install Python and protoc, etc. I would rather merge this, get CI going, and then improve it in a second pass, so let's delay that.